src/react/branching.js

```diff
@@ -39,78 +39,132 @@ export type BranchStatusEnum = "ROOT" | "NO_BRANCH" | "NEW_BRANCH" | "BRANCH";
 // objects depending on various heuristics (if they have the same "type" for example)
 // A new branch state is created on a branch status of "NEW_BRANCH" and is reset to null once the branch is no
```

Comment:
^^ this comment seems completely irrelevant now. Could you please make sure that when the code changes, the comments are still helpful?